staging/src/k8s.io/apiserver/pkg/server/patch_config.go

```diff
@@ -0,0 +1,49 @@
+package server
+
+import (
+	"context"
+	"net/http"
+	"strings"
+
+	"k8s.io/apiserver/pkg/authorization/authorizer"
+)
+
+// WithNotFoundProtectorHandler will return 429 instead of 404 iff:
+//  - server hasn't been ready (/readyz=false)
+//  - the user is GC or the namespace lifecycle controller
+//  - the path is for an aggregated API or CR
+//
+// This handler ensures that the system stays consistent even when requests are received before the server is ready.
+// In particular it prevents child deletion in case of GC or/and orphaned content in case of the namespaces controller.
+func WithNotFoundProtectorHandler(delegate http.Handler, hasBeenReadyCh <-chan struct{}, authorizerAttributesFunc func(ctx context.Context) (authorizer.Attributes, error)) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		select {
+		case <-hasBeenReadyCh:
+			delegate.ServeHTTP(w, r)
+			return
+		default:
+		}
+
+		ctx := r.Context()
+		attribs, err := authorizerAttributesFunc(ctx)
+		if err != nil {
+			delegate.ServeHTTP(w, r)
+			return
+		}
+
+		if patchMatches(r.URL.Path) && userMatches(attribs.GetUser().GetName()) {
+			w.Header().Set("Retry-After", "3")
+			http.Error(w, "Too many requests, please try again later.", http.StatusTooManyRequests)
```

Comment:
does this match the 429 returned by p&f?  I thought we tried to return json.

Can the message here be distinct so we can find it while tracing?

Reply:
> does this match the 429 returned by p&f? I thought we tried to return json

yes, [p&f](https://github.com/kubernetes/kubernetes/blob/6f8b01e2e891ac7374b1a66fc4b931b71a3be615/staging/src/k8s.io/apiserver/pkg/server/filters/priority-and-fairness.go#L182) uses [tooManyRequest](https://github.com/kubernetes/kubernetes/blob/6f8b01e2e891ac7374b1a66fc4b931b71a3be615/staging/src/k8s.io/apiserver/pkg/server/filters/maxinflight.go#L219) function

Reply:
> Can the message here be distinct so we can find it while tracing?

+1